// appium-flutter-driver/build/lib/doctor/checks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.optionalApplesimutilsCommandCheck = exports.optionalIdbCommandCheck = exports.envVarAndPathCheck = exports.xcodeToolsCheck = exports.xcodeCheck = exports.optionalGstreamerCheck = exports.optionalBundletoolCheck = exports.androidSdkCheck = exports.javaHomeValueCheck = exports.javaHomeCheck = exports.androidHomeCheck = exports.optionalFfmpegCheck = exports.homeEnvVarCheck = void 0;
const appium_android_driver_1 = require("appium-android-driver");
const appium_xcuitest_driver_1 = require("appium-xcuitest-driver");
// shared
exports.homeEnvVarCheck = appium_xcuitest_driver_1.doctor.required.homeEnvVarCheck;
let androidHomeCheck;
let javaHomeCheck;
let javaHomeValueCheck;
let androidSdkCheck;
let optionalBundletoolCheck;
let optionalGstreamerCheck;
if (!process.env.SKIP_ANDROID) {
    exports.androidHomeCheck = androidHomeCheck = appium_android_driver_1.doctor.androidHomeCheck;
    exports.javaHomeCheck = javaHomeCheck = appium_android_driver_1.doctor.javaHomeCheck;
    exports.javaHomeValueCheck = javaHomeValueCheck = appium_android_driver_1.doctor.javaHomeValueCheck;
    exports.androidSdkCheck = androidSdkCheck = appium_android_driver_1.doctor.androidSdkCheck;
    exports.optionalBundletoolCheck = optionalBundletoolCheck = appium_android_driver_1.doctor.optionalBundletoolCheck;
    exports.optionalGstreamerCheck = optionalGstreamerCheck = appium_android_driver_1.doctor.optionalGstreamerCheck;
}
let xcodeCheck;
let xcodeToolsCheck;
let envVarAndPathCheck;
let optionalIdbCommandCheck;
let optionalApplesimutilsCommandCheck;
if (!process.env.SKIP_IOS) {
    exports.xcodeCheck = xcodeCheck = appium_xcuitest_driver_1.doctor.required.xcodeCheck;
    exports.xcodeToolsCheck = xcodeToolsCheck = appium_xcuitest_driver_1.doctor.required.xcodeToolsCheck;
    exports.envVarAndPathCheck = envVarAndPathCheck = appium_xcuitest_driver_1.doctor.required.envVarAndPathCheck;
    exports.optionalIdbCommandCheck = optionalIdbCommandCheck = appium_xcuitest_driver_1.doctor.optional.optionalIdbCommandCheck;
    exports.optionalApplesimutilsCommandCheck = optionalApplesimutilsCommandCheck = appium_xcuitest_driver_1.doctor.optional.optionalApplesimutilsCheck;
}
// shared
exports.optionalFfmpegCheck = appium_android_driver_1.doctor.optionalFfmpegCheck;
